upload/fileUpload.go

```diff
@@ -71,14 +73,37 @@ func getGooglePhotosAlbumID(name string, c *gphotos.Client) string {
 	return album.Id
 }
 
-func (f *Item) upload(completedUploads *completeduploads.Service) error {
+func (f *Item) upload(completedUploads *completeduploads.Service, uploadURLsService *uploadurls.Service) error {
 	albumID := getGooglePhotosAlbumID(f.album, f.client)
 	log.Printf("Uploading object: file=%s", f.path)
 
+	// check upload URL db for previous uploads
+	log.Println("Looking up upload URLs database for ", f.path)
+	curUploadURL, err := uploadURLsService.GetUploadURL(f.path)
+	if err != nil {
+		// Not found, not an error, just an empty upload URL
+		log.Println(err)
```

Comment:
I will add here `curUploadURL = ""`. I will not trust on the response of `uploadURLsService.GetUploadURL)` in case of error.

Reply:
`GetUploadURL()` returns `""` in all error cases, but as you will.

Done.

Reply:
It returns now... but eventually could return another thing... when a function gives an error, we should not trust on the returned value. :+1